agent/consul/fsm/commands_oss_test.go

```diff
@@ -1397,3 +1401,140 @@ func TestFSM_ConfigEntry(t *testing.T) {
 		require.Equal(entry, config)
 	}
 }
+
+// This adapts another test by chunking the encoded data and then performing
+// out-of-order applies of half the logs. It then snapshots, restores to a new
+// FSM, and applies the rest. The goal is to verify that chunking snapshotting
+// works as expected.
+func TestFSM_Chunking_Lifecycle(t *testing.T) {
+	t.Parallel()
+	fsm, err := New(nil, os.Stderr)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	req := structs.RegisterRequest{
+		Datacenter: "dc1",
+		Node:       "foo",
+		Address:    "127.0.0.1",
+		Service: &structs.NodeService{
+			ID:      "db",
+			Service: "db",
+			Tags:    []string{"master"},
+			Port:    8000,
+		},
+		Check: &structs.HealthCheck{
+			Node:      "foo",
+			CheckID:   "db",
+			Name:      "db connectivity",
+			Status:    api.HealthPassing,
+			ServiceID: "db",
+		},
+	}
+	buf, err := structs.Encode(structs.RegisterRequestType, req)
+	if err != nil {
+		t.Fatalf("err: %v", err)
+	}
+
+	var opNum uint64 = 32
+	var logs []*raft.Log
+	for i, b := range buf {
```

Comment:
So this is writing each byte of the serialized `RegisterRequest` as a separate raft log entry right?

Reply:
Yep.